Sources/Plasma/FeatureLib/pfConditional/plObjectInBoxConditionalObject.cpp

```diff
@@ -286,163 +290,19 @@ void plVolumeSensorConditionalObject::Read(hsStream* stream, hsResMgr* mgr)
-                                {   
-                                    if (plNetClientApp::GetInstance()->GetLocalPlayerKey() != pActivateMsg->fHitterObj)
-                                    {
-                                        plArmatureMod *am=const_cast<plArmatureMod*>( plArmatureMod::ConvertNoRef(pObj->GetModifier(i)));
-                                        if (!am->IsLocalAI())
-                                        {
-                                            return false;
-                                        }
-                                    }
-                                }
-                            }
-                            plSynchedObject* syncObj = (plSynchedObject*)pObj;
-                            if (syncObj->IsLocallyOwned() != plSynchedObject::kYes) 
-                            {
-                                return false;
-                            }
-                        }
-                    }
-                    if (fType == kTypeExit)
-                    {
-                        fLogicMod->GetNotify()->AddCollisionEvent(false, pActivateMsg->fHitterObj, pActivateMsg->fHiteeObj, false);
-                        //fLogicMod->RequestTrigger(false);
-                        if (!Satisfied())
-                            return false;
-                    
-                        fLogicMod->Trigger(false);
-                    }
-                    return false;
-                }
-            }
-        }
 
-        return false;
-    }
-
-    plPlayerPageMsg* page = plPlayerPageMsg::ConvertNoRef(msg);
-    if(page && page->fUnload)
-    {
-        for(int j= 0; j< fInside.Count(); j++)
-        {
-            if(fInside[j] == page->fPlayer)
-            {//this is the one inside
-                if(fHittee)
-                {
-                    plSceneObject *so = plSceneObject::ConvertNoRef(fHittee->ObjectIsLoaded());
-                    if(so && so->IsLocallyOwned())
-                    {
-                        if (fType == kTypeExit)
-                        {
-                            fLogicMod->GetNotify()->AddCollisionEvent(false, page->fPlayer, fHittee, false);
-                            //fLogicMod->RequestTrigger(false);
-                            if (!Satisfied())
-                                return false;
-                            fLogicMod->Trigger(false);
-                        }
-                    }
-                }
-                fInside.Remove(j);
-            }
-        }
-    }
-        return plConditionalObject::MsgReceive(msg);
-}
 void plVolumeSensorConditionalObjectNoArbitration::Read(hsStream* stream, hsResMgr* mgr)
 {
     plVolumeSensorConditionalObject::Read(stream, mgr);
-    plgDispatch::Dispatch()->RegisterForExactType(plPlayerPageMsg::Index(), GetKey());
+
+    // We must have a valid fpKey before we do this, hence why this is not in the constructor
```

Comment:
Ah. :trollface: I thought it was a local member that I wasn't seeing... or a typo.